### DISCUSSION TITLE
Router parameter naming style conversion

### DISCUSSION THREAD

**Discussion by tbontb-iaq** [posva/unplugin-vue-router#468] [Ideas]
Currently, if I have `pages/user/[u-id]/group/[g-id].vue`, it will generate params like:

```ts
export interface RouteNamedMap {
  "/user/[u-id]/group/[g-id]": RouteRecordInfo<
    "/user/[u-id]/group/[g-id]",
    "/user/:u-id/group/:g-id",
    { u-id: ParamValue<true>; g-id: ParamValue<true> },
    { u-id: ParamValue<false>; g-id: ParamValue<false> }
  >;
}
```

And `u-id` and `g-id` are not valid JavaScript attribute names.

Is it possible to add some kind of automatic conversion? It should also be configurable to convert to multiple naming styles, like camelCase, underscores or raw style.

**Comment by posva** [comment]
You can indeed automatically convert them by using the `extendRoute` option. Something like this:

```ts
      extendRoute(route) {
                // transform kebab-case to camelCase
        route.params.forEach((param, i) => {
          const newParam = param.paramName.replace(/-([a-z])/g, (g) =>
            g[1].toUpperCase()
          )
          route.path = route.path.replace(`:${param.paramName}`, `:${newParam}`)
          param.paramName = newParam
        })
	  }
```

But also, the generated TS should be correct, so it should be `'g-id'`.

**Comment by tbontb-iaq** [comment]
Thanks for your answer. Will this become a built-in feature?

**Comment by posva** [comment]
Right now I don’t think it needs a dedicated option

**Comment by tbontb-iaq** [comment]
`question/[q-id]` will be converted to `question/:q-id`, which will match `question/1234-id` instead of `question/1234`. Any good ideas?

**Comment by posva** [comment]
I updated the function, it does require you to change the path too.

**Comment by tbontb-iaq** [comment]
Not work, but got warning: `[vite] ⚠️  [unplugin-vue-router]: Only absolute paths are supported. Make sure that ":q_id" starts with a slash "/".`

**Comment by posva** [comment]
That's #341, so you will need a bit more code to handle it right now

**Comment by posva** [comment]
I'm still not convinced at all for the need of an option to handle this. Naming the files so they match vue router params seems like a good practice

**Reply by abdul-alhasany** [reply]
Technically it is working. But the usual naming convection of files is `kebab-case` while js variable names are `camelCase`. It will play nice with ESLint and provide better developer experince.
